### DISCUSSION TITLE
关于engine单线程的问题

### DISCUSSION THREAD

**Discussion by yangqingyuan** [alibaba/xquic#81] [Q&A] [answered]
文档中有提到xqc_engine_create创建出来的engine内部是单线程的，允许单个进程/或者不同的进程中创建多个engine
问题如下：
（1）单线程是不是意味着，多个stream（不同线程）用同一个engine同时向服务队发送数据的时候是串行的，排队执行？
（2）如果要engine并发传输，该怎么做，创建多个engine，不同的stream用不同的engine？

场景：
例如客户端封装了两个接口，通过xqc_client_stream_send或者xqc_client_request_send的方式想服务端发送请求数据，不希望第一个接口执行完毕后再执行第二个接口

**Comment by Kulsk** [comment, marked as answer]
sending or processing data is always one by one within an xqc engine, but multiple engines can be created and be used in multiple threads or processes, with one-to-one corresponding relation, just remember that one engine must not be accessed by miltiple threads or processes at the same time.
